backend/app/controllers/spree/admin/store_credits_controller.rb

```diff
@@ -38,6 +92,28 @@ def load_categories
       def set_action_originator
         @object.action_originator = try_spree_current_user
       end
+
+      def ensure_update_reason
+        @update_reason = Spree::StoreCreditUpdateReason.find_by(id: params[:update_reason_id])
+        unless @update_reason
+          @store_credit.errors.add(:base, Spree.t("admin.store_credits.errors.update_reason_required"))
+          render_edit_page
+        end
+      end
+
+      def render_edit_page
+        if action == :update_amount
```

Comment:
The advantage I see is centralizing the error handling for both actions. Since there are three operations going on here (loading the update reasons, setting the flash error message and rendering a template) I think centralizing the logic is worthwhile but happy to remove the method if more people feel the same.